curator-recipes/src/main/java/org/apache/curator/framework/recipes/cache/PathChildrenCache.java

```diff
@@ -694,14 +695,23 @@ private void applyNewData(String fullPath, int resultCode, Stat stat, byte[] byt
             {
                 offerOperation(new EventOperation(this, new PathChildrenCacheEvent(PathChildrenCacheEvent.Type.CHILD_ADDED, data)));
             }
-            else if ( previousData.getStat().getVersion() != stat.getVersion() )
+            else if ( hasChanged(stat, previousData, data) )
             {
                 offerOperation(new EventOperation(this, new PathChildrenCacheEvent(PathChildrenCacheEvent.Type.CHILD_UPDATED, data)));
             }
             updateInitialSet(ZKPaths.getNodeFromPath(fullPath), data);
         }
     }
 
+    private boolean hasChanged(Stat stat, ChildData previousData, ChildData newData)
+    {
+        if ( cacheData )
+        {
+            return !Arrays.equals(previousData.getData(), newData.getData());
```

Comment:
Note: TreeCache uses only Mzxid to determine if there's an update. Is that good enough?